src/modm/platform/gpio/stm32/software_port.hpp.in

```diff
@@ -74,11 +75,13 @@ public:
 	%% endfor
 		}
 %% endfor
+%% endif
 		return r;
 	}
 
 	static void write(PortType data)
```

Comment:
The difficulty here is with a case like this:

```c++
inline static void
setInputTrigger(const InputTrigger trigger)
{
    %% if not is_mock
    switch (trigger)
    {
    case InputTrigger::RisingEdge:
        EXTI->RTSR |=  mask;
        EXTI->FTSR &= ~mask;
        break;
    case InputTrigger::FallingEdge:
        EXTI->RTSR &= ~mask;
        EXTI->FTSR |=  mask;
        break;
    case InputTrigger::BothEdges:
        EXTI->RTSR |=  mask;
        EXTI->FTSR |=  mask;
        break;
    }
    %% endif
}
```

I can't remove the parameter name because the parameter is needed for non-mocks. I could more aggressively `%% if` out the method definition, but I was somewhat unhappy with that since it's so inconsistent. So I'm not sure what makes the most sense here. Maybe the best option really is just to have two copies of the method signature.

Note that there's one file, `set.cpp`, which I created an entirely new mock file for because it's pretty much all different.

Reply:
I kinda think just having two copies of the method signature would be fine. It would mean a bit more `%% if`s, but would be better than creating whole new files since the structure for each build platform should be the same for this file.